python/Ganga/Utility/execute.py

```diff
@@ -230,10 +228,10 @@ def execute(command,
 
     if update_env:
         env_output_key = 'env_output'
-        update_env_thread = update_thread(env_file_pipes, thread_output, env_output_key)
+        update_env_thread = update_thread(env_file_pipes, thread_output, env_output_key, require_output=True)
     if not shell:
         pkl_output_key = 'pkl_output'
-        update_pkl_thread = update_thread(pkl_file_pipes, thread_output, pkl_output_key)
+        update_pkl_thread = update_thread(pkl_file_pipes, thread_output, pkl_output_key, require_output=False)
```

Comment:
If `shell=False` then we must always *allow* for some pickled output to be sent through the stream (either from `output()` or from a caught exception). We expect that there *might* be some but also there *might not*. If there *is* then we unpickle it and end up returning it from `execute()`. We don't *require* that there is some output.

We can't, from within `execute()` know, a priori, which `command`s will be pushing though the pickle stream and which will simply be printing it through stdout.